pkg/apis/management.cattle.io/v3/cluster_types.go

```diff
@@ -108,21 +108,22 @@ type Cluster struct {
 }
 
 type ClusterSpecBase struct {
-	DesiredAgentImage                    string                                  `json:"desiredAgentImage"`
-	DesiredAuthImage                     string                                  `json:"desiredAuthImage"`
-	AgentImageOverride                   string                                  `json:"agentImageOverride"`
-	AgentEnvVars                         []v1.EnvVar                             `json:"agentEnvVars,omitempty"`
-	RancherKubernetesEngineConfig        *rketypes.RancherKubernetesEngineConfig `json:"rancherKubernetesEngineConfig,omitempty"`
-	DefaultPodSecurityPolicyTemplateName string                                  `json:"defaultPodSecurityPolicyTemplateName,omitempty" norman:"type=reference[podSecurityPolicyTemplate]"`
-	DefaultClusterRoleForProjectMembers  string                                  `json:"defaultClusterRoleForProjectMembers,omitempty" norman:"type=reference[roleTemplate]"`
-	DockerRootDir                        string                                  `json:"dockerRootDir,omitempty" norman:"default=/var/lib/docker"`
-	EnableNetworkPolicy                  *bool                                   `json:"enableNetworkPolicy" norman:"default=false"`
-	EnableClusterAlerting                bool                                    `json:"enableClusterAlerting" norman:"default=false"`
-	EnableClusterMonitoring              bool                                    `json:"enableClusterMonitoring" norman:"default=false"`
-	WindowsPreferedCluster               bool                                    `json:"windowsPreferedCluster" norman:"noupdate"`
-	LocalClusterAuthEndpoint             LocalClusterAuthEndpoint                `json:"localClusterAuthEndpoint,omitempty"`
-	ScheduledClusterScan                 *ScheduledClusterScan                   `json:"scheduledClusterScan,omitempty"`
-	ClusterSecrets                       ClusterSecrets                          `json:"clusterSecrets" norman:"nocreate,noupdate"`
+	DesiredAgentImage                                  string                                  `json:"desiredAgentImage"`
+	DesiredAuthImage                                   string                                  `json:"desiredAuthImage"`
+	AgentImageOverride                                 string                                  `json:"agentImageOverride"`
+	AgentEnvVars                                       []v1.EnvVar                             `json:"agentEnvVars,omitempty"`
+	RancherKubernetesEngineConfig                      *rketypes.RancherKubernetesEngineConfig `json:"rancherKubernetesEngineConfig,omitempty"`
+	DefaultPodSecurityAdmissionConfigurationTemplateId string                                  `json:"defaultPodSecurityAdmissionConfigurationTemplateId,omitempty"`
```

Comment:
I know it's in the requirements of the linked issue, but would it make more sense to name it `defaultPodSecurityAdmissionConfigurationTemplateName` to match the `defaultPodSecurityPolicyTemplateName` (IMHO it better relays the fact we reference Name and not some sort of ID).

Though I see `defaultPodSecurityPolicyTemplateId` in a few (autogenerated) files in Rancher so not sure why is that. But I clearly see `defaultPodSecurityPolicyTemplateName` on `cluster.provisioning.cattle.io` objects.

Reply:
Yeah, I can make this change - a lot of the naming conventions were taken directly from the spec but having consistency between the fields would look better. The auto-generated files contain `id` simply because this field contains `id`